Algorithm-Pairwise.md

```diff
@@ -44,16 +44,18 @@ It is easy to confuse indices as being numbers, but since you will be interactin
 function pairwise(arr, arg) {
  // Set sum of indices to zero
  var sum = 0;
+ // make a local copy of the arguments object so we don't modify it directly
+ var pairArr = arr;
```

Comment:
This still mutates the original array. You are creating a new reference pointer that still points to the same array as `arr`.

You need to create a new Array `pairArr`, using the values in `arr`. Look into `Array.prototype.from()` or `Array.prototype.slice()`.